rclpy/src/rclpy/context.cpp

```diff
@@ -149,13 +152,20 @@ Context::shutdown()
 {
   {
     std::lock_guard guard{g_contexts_mutex};
-    g_contexts.erase(
-      std::remove(g_contexts.begin(), g_contexts.end(), rcl_context_.get()),
-      g_contexts.end());
+    if (std::find(g_contexts.begin(), g_contexts.end(), rcl_context_.get()) != g_contexts.end()) {
+      g_contexts.erase(
+        std::remove(g_contexts.begin(), g_contexts.end(), rcl_context_.get()),
+        g_contexts.end());
+    }
   }
-  rcl_ret_t ret = rcl_shutdown(rcl_context_.get());
-  if (RCL_RET_OK != ret) {
-    throw RCLError("failed to shutdown");
+  // Just in case, check if context is valid
+  if (rcl_context_is_valid(rcl_context_.get())) {
```

Comment:
`rcl_shutdown` already does this check, so if we're going to throw anyways then we don't need this check here.

https://github.com/ros2/rcl/blob/493ebf3a5d8fe2d861a93006a7d6908fd4e592d9/rcl/src/rcl/init.c#L241

Reply:
okay, i will remove this check.